compiler/rustc_typeck/src/check/fn_ctxt/_impl.rs

```diff
@@ -313,6 +313,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     ) => {
                         // A reborrow has no effect before a dereference.
                     }
+                    (&[Adjustment { kind: Adjust::Deref(None), .. }], _) => {}
```

Comment:
This path is still a bug, right? It should never be hit unless the compiler will error later. If so, please add a delay_span_bug call and leave a comment explaining why this can happen

Reply:
Should it be a bug? I never used delay_span_bug so I don't know the difference between bug! and delay_span_bug 😸

Reply:
ah, well, `delay_span_bug` is a safe-guard that only ICEs the compiler if no other errors have been reported.

Reply:
I added `delay_span_bug` as you suggested it

Reply:
Thanks